File: frontend/src/api/types/leaderAi.ts
export type ApiWeeklyBriefing = {
  constituency_name: string
  period_start: string
  period_end: string
  headline: string
  summary: string
  highlights: string[]
  risks: string[]
  recommendations: string[]
}

export type ApiPriorityInsightItem = {
  id: string
  explanation: string
  recommended_action: string
}

export type ApiPriorityInsightsResponse = {
  overview: string
  items: ApiPriorityInsightItem[]
}

export type ApiComplaintTheme = {
  theme: string
  category: string
  count: number
  summary: string
  severity: string
  wards: string[]
}

export type ApiComplaintThemesResponse = {
  overview: string
  period_label: string
  total_complaints: number
  themes: ApiComplaintTheme[]
}

File: backend/app/schemas/leader_ai.py
from pydantic import BaseModel


class WeeklyBriefingResponse(BaseModel):
    constituency_name: str
    period_start: str
    period_end: str
    headline: str
    summary: str
    highlights: list[str]
    risks: list[str]
    recommendations: list[str]


class PriorityInsightItem(BaseModel):
    id: str
    explanation: str
    recommended_action: str


class PriorityInsightsResponse(BaseModel):
    overview: str
    items: list[PriorityInsightItem]


class ComplaintTheme(BaseModel):
    theme: str
    category: str
    count: int
    summary: str
    severity: str
    wards: list[str]


class ComplaintThemesResponse(BaseModel):
    overview: str
    period_label: str
    total_complaints: int
    themes: list[ComplaintTheme]
